mu/app.py

```diff
@@ -275,15 +284,17 @@ class SharedMemoryMutex(object):
     NAME = "mu-tex"
 
     def __init__(self):
-        self._shared_memory = QSharedMemory(self.NAME)
+        sharedAppName = self.NAME
+        if "MU_TEST_SUPPORT_RANDOM_APP_NAME_EXT" in os.environ:
+            sharedAppName += os.environ["MU_TEST_SUPPORT_RANDOM_APP_NAME_EXT"]
+        self._shared_memory = QSharedMemory(sharedAppName)
```

Comment:
During automated tests a test failure would sometimes leave the shared memory block in place and the developer had to reboot their computer to be able to run tests of handlers to block multiple instances from running.  This prevents that by - only during automated tests - using a random name for the app as the key for the shared memory block so subsequent test runs won't conflict with each other in the case where a prior one crashes without cleaning up.